src/TriggerBinding/SqlTriggerListener.cs

```diff
@@ -0,0 +1,364 @@
+
+                    using (SqlTransaction transaction = connection.BeginTransaction(System.Data.IsolationLevel.RepeatableRead))
+                    {
+                        await CreateSchemaAsync(connection, transaction, cancellationToken);
+                        await CreateGlobalStateTableAsync(connection, transaction, cancellationToken);
+                        await this.InsertGlobalStateTableRowAsync(connection, transaction, userTableId, cancellationToken);
+                        await CreateWorkerTablesAsync(connection, transaction, workerTableName, primaryKeyColumns, cancellationToken);
+                        transaction.Commit();
+                    }
+
+                    // TODO: Check if passing cancellation token would be beneficial.
+                    this._changeMonitor = new SqlTableChangeMonitor<T>(
+                        this._connectionString,
+                        userTableId,
+                        this._userTable,
+                        this._userFunctionId,
+                        workerTableName,
+                        userTableColumns,
+                        primaryKeyColumns.Select(col => col.name).ToList(),
+                        this._executor,
+                        this._logger);
+
+                    this._state = State.Running;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Stops the listener (if it was started), which stops checking for changes on the user's table.
+        /// </summary>
+        public Task StopAsync(CancellationToken cancellationToken)
+        {
+            // Nothing to stop if the change monitor has either already been stopped or hasn't been started.
+            if (this._state == State.Running)
+            {
+                this._changeMonitor.Stop();
+                this._state = State.Stopped;
+            }
+            return Task.CompletedTask;
+        }
+
+        /// <summary>
+        /// Returns the OBJECT_ID of userTable
+        /// </summary>
+        /// <exception cref="InvalidOperationException">
+        /// Thrown if the query to retrieve the OBJECT_ID of the user table fails to correctly execute
+        /// This can happen if the OBJECT_ID call returns NULL, meaning that the user table might not exist in the database
+        /// </exception>
+        private async Task<int> GetUserTableIdAsync(SqlConnection connection, CancellationToken cancellationToken)
+        {
+            string getObjectIdQuery = $"SELECT OBJECT_ID(N{this._userTable.QuotedName}, 'U');";
+
+            using (var getObjectIdCommand = new SqlCommand(getObjectIdQuery, connection))
+            {
+                using (SqlDataReader reader = await getObjectIdCommand.ExecuteReaderAsync(cancellationToken))
+                {
+
+                    // TODO: Check if the below if-block ever gets hit.
```

Comment:
Has this TODO been done?

It shouldn't ever be hit, but doesn't hurt to be explicit. Would prefer a better error message though so it's clear what the problem is if a user files an issue with this message.

Reply:
I would prefer to not have this additional condition. I tried a couple of ways to hit the if-block, but it was either `connection.Open()` or `ExecuteReaderAsync()` that failed. There are a few other locations in the repo where `ReadAsync` method gets called but they do not get special treatment. If the method fails in any case, we would still be encountering exception, just not the one of our choice.

Reply:
On second thought, we should be taking into account all return values before moving ahead. I will update the exception message.